cgi-bin/DW/Console/Command/BonusIcons.pm

```diff
@@ -85,27 +85,26 @@ sub execute {
         }
 
     } elsif ( $cmd eq 'xfer' ) {
-        my $target_u = LJ::load_user( $cmdarg );
+        my $destination_u = LJ::load_user( $cmdarg );
 
         return $self->error( 'Invalid target user.' )
-            unless $target_u;
+            unless $destination_u;
         return $self->error( 'E-mail addresses do not match.' )
-            unless $to_u->prop( 'email' ) eq $target_u->prop( 'email' ); 
+            unless $to_u->prop( 'email' ) eq $destination_u->prop( 'email' ); 
+        return $self->error( 'E-mail address is not confirmed.' )
+            unless $to_u->is_validated && $destination_u->is_validated;
 
         my $xfer_count = $to_u->prop( 'bonus_icons' );
-        my $new_to_u = max( $to_u->prop( 'bonus_icons' ) - $xfer_count, 0 );
-        $to_u->set_prop( bonus_icons => $new_to_u );
+        $to_u->set_prop( bonus_icons => 0 );
         LJ::statushistory_add( $to_u, $remote, 'bonus_icons',
                 sprintf( 'Transferred %d icons to %s.', $xfer_count,
-                    $target_u->user ) );
-        my $new_target_u = max( $target_u->prop( 'bonus_icons' ) 
-            + $xfer_count, 0 );
-        $target_u->set_prop( bonus_icons => $new_target_u );
-        LJ::statushistory_add( $target_u, $remote, 'bonus_icons',
+                    $destination_u->user ) );
+        $destination_u->set_prop( bonus_icons => $destination_u->prop( 'bonus_icons' ) + $xfer_count );
```

Comment:
Because you're referencing this number so often, and because the `prop` method has a certain amount of computational overhead, I would still save this to a variable here:
```
my $new_total = $destination_u->prop( 'bonus_icons' ) + $xfer_count;
$destination_u->set_prop( bonus_icons => $new_total );
```
And then in the two places below where you call `$destination_u->prop( 'bonus_icons' )` you can change those to `$new_total` as well.  Or whatever you want to call it; it was $new_target_u before but you renamed $target_u, so... :grin: 

The rest is much easier to read now, thanks! :+1: